Tests/SwiftSyntaxMacroExpansionTest/MacroReplacementTests.swift

```diff
@@ -113,4 +114,52 @@ final class MacroReplacementTests: XCTestCase {
       """
     )
   }
+
+  func testMacroGenericArgumentExpansion() throws {
+    let macro: DeclSyntax =
+      """
+      macro gen<A, B>(a: A, b: B) = #otherMacro<A, B>(first: a, second: b)
+      """
+
+    let use: ExprSyntax =
+      """
+      #gen<Int, String>(a: 5, b: "Hello")
+      """
+
+    let macroDecl = macro.as(MacroDeclSyntax.self)!
+    let definition = try macroDecl.checkDefinition()
+    guard case let .expansion(expansion, replacements, genericReplacements) = definition else {
+      XCTFail("not a normal expansion")
+      fatalError()
+    }
+
+    guard let replacementA = genericReplacements.first else {
+      XCTFail("Expected generic replacement for A")
+      fatalError()
+    }
+    guard let replacementB = genericReplacements.dropFirst().first else {
+      XCTFail("Expected generic replacement for A")
+      fatalError()
+    } 
+    XCTAssertEqual(genericReplacements.count, 2)
+
+    XCTAssertEqual(replacementA.parameterIndex, 0)
+    XCTAssertEqual("\(replacementA.reference.argument)", "A")
+
+    XCTAssertEqual(replacementB.parameterIndex, 1)
+    XCTAssertEqual("\(replacementB.reference.argument)", "B")
+
+    let expandedSyntax = macroDecl.expand(
+      use.as(MacroExpansionExprSyntax.self)!,
+      definition: expansion,
+      replacements: replacements,
+      genericReplacements: genericReplacements
+    )
+    assertStringsEqualWithDiff(
+      expandedSyntax.description,
+      """
+      #otherMacro<A, B>(first: 5, second: "Hello")
```

Comment:
Am I missing something or should `A` and `B` be replaced by `Int` and `String` so this would be

```swift
#otherMacro<Int, String>(first: 5, second: "Hello")
```

Reply:
I think the actual replacing only happens in Swift -- at least that's where the code to handle replacing the text is. I was confused by this actually, but this seems to work in Swift itself then 🤔

Reply:
Aha Ben explained this to me -- it seems we have two macro expansion mechanisms huh... I'll implement it here too then

Reply:
Implemented it over here as well

I was a bit confused about the trailing trivia attached... I left a TODO and would appreciate someone helping out on that bit.

@bnbarham